App.js

```diff
@@ -0,0 +1,506 @@
+  }
+
+  function fun(){
+    console.log("fun loaded");
+  }
+
+
+  return (
+    <SideMenu
+      menu={menu}
+      isOpen={isOpen}
+      onChange={() => doSomething(isOpen)}
+      menuPosition={'left'}
+      scrollsToTop={false}>
+      <SafeAreaView
+        style={(backgroundStyle, {backgroundColor: nMode ? 'black' : 'white'})}>
+        <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
+        {kAwake && <KeepAwake />}
+        <View style={styles.hdr}>
+          <View style={{width: '33%'}}>
+            <TouchableOpacity
+              style={{marginLeft: 10}}
+              onPress={() => doSomething()}>
+              <Image
+                source={require('./icons/menuicon.png')}
+                style={{width: 40, height: 40}}
+              />
+            </TouchableOpacity>
+          </View>
+          <View style={({width: '33%'}, styles.hdr)}>
+            <TouchableOpacity onPress={() => ang > 0 && getData(+ang - 1)}>
+              <Image
+                source={require('./icons/left_arw.png')}
+                style={{width: 30, height: 25}}
+              />
+            </TouchableOpacity>
+            <TouchableOpacity onPress={() => setAngModal(true)}>
+              <Text style={{padding:5, paddingHorizontal:10, borderColor:'#ccc', borderWidth:1, marginHorizontal:5}}>{ang}</Text>
+            </TouchableOpacity>
+            
+            <TouchableOpacity onPress={() => getData(+ang + 1)}>
+              <Image
+                source={require('./icons/right_arw.png')}
+                style={{width: 30, height: 25}}
+              />
+            </TouchableOpacity>
+          </View>
+          <View style={{width: '33%', alignItems: 'flex-end'}}>
+            <Image
+              source={require('./icons/toggle.jpg')}
+              style={{width: 35, height: 35}}
+            />
+          </View>
+        </View>
+
+        <ScrollView
+          contentInsetAdjustmentBehavior="automatic"
+          style={backgroundStyle}>
+          <GestureRecognizer
+            onSwipeLeft={() => canSwipe && getData(+ang + 1)}
+            onSwipeRight={() => canSwipe && ang > 0 && getData(+ang - 1)}
+            style={{
+              flex: 1,
+            }}>
+            <View
+              style={{
+                backgroundColor: nMode ? 'black' : 'white',
+              }}>
+               
+              <Section>
+                <View
+                  style={{
+                    flexDirection: 'row',
+                    flexWrap: 'wrap',
+                    backgroundColor: nMode ? 'black' : 'white',
+                  }}>
+                  <Text
+                    style={{color: nMode ? 'white' : 'black', fontSize: fSize}}>
+                    {resData}
+                  </Text>
+                </View>
+              </Section>
+            </View>
+          </GestureRecognizer>
+          
+        </ScrollView>
+        
+        <Modal animationType="slide"
```

Comment:
We are using three times `<Modal />` here. Can we use just one Modal and just update the content asynchronously?

Reply:
major rework